messenger/messenger.go

```diff
@@ -229,30 +215,29 @@ func (m *MesosMessenger) Route(ctx context.Context, upid *upid.UPID, msg proto.M
 	return m.tr.Inject(ctx, &Message{upid, name, msg, data})
 }
 
-// Start starts the messenger.
+// Start starts the messenger; expects to be called once and only once.
```

Comment:
This code is ripe for refactoring. Not in this PR.

On Thu, Sep 24, 2015 at 5:49 AM, Tomás Senart <notifications@github.com>
wrote:

> In messenger/messenger.go
> <https://github.com/mesos/mesos-go/pull/189#discussion_r40302068>:
>
> > @@ -229,30 +215,29 @@ func (m *MesosMessenger) Route(ctx context.Context, upid *upid.UPID, msg proto.M
> >  	return m.tr.Inject(ctx, &Message{upid, name, msg, data})
> >  }
> >
> > -// Start starts the messenger.
> > +// Start starts the messenger; expects to be called once and only once.
>
> https://golang.org/pkg/sync/#Once ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/189/files#r40302068>.
>